### ISSUE TITLE
nan loss 

### ISSUE THREAD

**Issue by dashings** [yijingru/BBAVectors-Oriented-Object-Detection#6]
When I train the model on the  DOTA dataset,  the value of "hm loss" is NAN  after 40 epochs. Do you know what is wrong?
![image](https://user-images.githubusercontent.com/13343070/91637579-146f5780-ea3c-11ea-8107-48f87cfc5d45.png)

And where can I find the "hrsc_evaluation_task1" file referred in the dataset_hrsc.py?
Thank you,Best wish.

**Comment by yijingru** [comment]
I've met this problem before. It could be the focal loss problem. I think one possible reason is the `log (x)` part, where the x is extremely approaching 0. I think the output of the heat map would have a very large negative value that after `sigmoid` it is 0. I think adding a Normalization layer in the second layer of the head part would help. But I didn't have this problem after trying large batch size. Maybe decreasing the learning rate would help too.

The `hrsc_evaluation_task1.py` is modified from `dota_evaluation_task1.py` in `DOTA_Devkit`, I uploaded it to the datasets, the code is similar to the original one.

**Comment by dashings** [comment]
> I've met this problem before. It could be the `focal loss` problem. I think one possible reason is the `log (x)` part, where the x is extremely approaching 0. I think the output of the heat map would have a very large negative value that after `sigmoid` it is 0. I think adding a Normalization layer in the second layer of the head part would help. But I didn't have this problem after trying large batch size. Maybe decreasing the learning rate would help too.
> 
> The `hrsc_evaluation_task1.py` is modified from `dota_evaluation_task1.py` in `DOTA_Devkit`, I uploaded it to the datasets, the code is similar to the original one.

Thank you very much. I would try.
